interp/constrextern.ml

```diff
@@ -274,17 +274,8 @@ let pattern_printable_in_both_syntax (ind,_ as c) =
 
  (* Better to use extern_glob_constr composed with injection/retraction ?? *)
 let rec extern_cases_pattern_in_scope (scopes:local_scopes) vars pat =
-  (* pboutill: There are letins in pat which is incompatible with notations and
-     not explicit application. *)
-  match pat with
-    | PatCstr(loc,cstrsp,args,na)
-	when !Flags.in_debugger||Inductiveops.constructor_has_local_defs cstrsp ->
-      let c = extern_reference loc Id.Set.empty (ConstructRef cstrsp) in
-      let args = List.map (extern_cases_pattern_in_scope scopes vars) args in
-      CPatCstr (loc, c, Some (add_patt_for_params (fst cstrsp) args), [])
-    | _ ->
   try
-    if !Flags.raw_print || !print_no_symbol then raise No_match;
+    if !Flags.in_debugger || !Flags.raw_print || !print_no_symbol then raise No_match;
```

Comment:
Having independent printers does not look dangerous to me. But reusing the existing printers does not look dangerous either to me. By reusing the existing printers, we can reuse the formatting rules without having to rewrite them from scratch. In the existing printers, there is already support for `Set Printing All`, so, why not just to reuse this existing work rather than rewriting a different printer.

By the way, there is actually an independent constr printer. Do `install_printer Top_printers.print_pure_constr` to install it. Maybe a first step to explore the direction that you are proposing would be to improve this independent constr printer to see if you can make it filling your needs.